src/app/pages/datasets/components/dataset-management/dataset-management.component.html

```diff
@@ -29,9 +29,35 @@ <h4 mat-card-title>Note: This is a work in progress for the new 'Dataset Managem
             routerLinkActive="selected"
           >
             <button mat-icon-button disabled></button>
-            <ix-dataset-node [dataset]="dataset"></ix-dataset-node>
+            <ix-dataset-node [dataset]="dataset | cast"></ix-dataset-node>
           </ix-tree-node>
 
+          <ix-nested-tree-node
+            *ixTreeNodeDef="let dataset; dataSource: dataSource, when: isDatasetRoot"
+            [ixTreeNodeDefDataSource]="dataSource"
+          >
+            <div
+              class="dataset-nested-tree-root-node"
+              (click)="onRowGroupSelected(dataset, $event)"
+            >
+              <ix-dataset-group-node [dataset]="dataset | cast">
```

Comment:
`ix-dataset-group-node` and `ix-vdev-group-node` looks the same. I'd suggest making it reusable to avoid creating `ix-something-group-node` in the future. The common group node component should be part of the IxTree module.

How it might look like:
![image](https://user-images.githubusercontent.com/351613/180015284-bb30d60e-cbfc-49d2-9e5e-4403459b0842.png)

Reply:
I'm okay with having separate components. They are similar, but have slightly different responsibilities.

Reply:
Each of `ix-dataset-group-node` and `ix-vdev-group-node` are presenting the data from their underlying data node, so it's good in terms of separation of responsibilities.